File: 1021-distribute-coins-in-binary-tree/1021-distribute-coins-in-binary-tree.cpp
class Solution {
public:
  
    int distributeCoins(TreeNode* root) {
           int moves=0;
         distribute(root,moves);
         return moves;
    }
     int distribute(TreeNode* root,int &moves){
        if(root==NULL) return 0;
        int  l=distribute(root->left,moves);
        int r=distribute(root->right,moves);
        moves+=abs(l)+abs(r);
        return (l+r+root->val)-1;
     }
};

File: 1021-distribute-coins-in-binary-tree/1021-distribute-coins-in-binary-tree.java
class Solution {
    int moves = 0;
    public int distributeCoins(TreeNode root) {

        
        distribute(root);
        return moves;
    }

    int distribute(TreeNode root){
        if(root==null) return 0;
        int  l=distribute(root.left);
        int r=distribute(root.right);
        moves+=Math.abs(l)+Math.abs(r);
        return (l+r+root.val)-1;
     }
}
